src/synapse-plugins/calculator-plugin.vala

```diff
@@ -27,15 +27,21 @@ namespace Synapse {
         public void activate () { }
         public void deactivate () { }
 
-        private class Result: Synapse.Match {
+        private class Result: Synapse.Match, Synapse.TextMatch {
             public int default_relevancy { get; set; default = 0; }
 
+            public string text { get; construct set; default = ""; }
+            public Synapse.TextOrigin text_origin { get; set; }
+
             public Result (double result, string match_string) {
                 Object (match_type: MatchType.TEXT,
+                        text: "%s = %g".printf (match_string, result),
```

Comment:
You may want to set this to a `Result.tooltip_text` or something similar and set `Result.text: "%g".printf (result)` so the result (`Result.text`) (and not the text) is copied instead.

Reply:
I've just realised this sounds incredibly confusing.

What's being copied is `Result.text`, so set `Result.text` to be the actual result ("4"), and set `Result.description` to be what you want to show on the tooltip (2+2 = 4)

Reply:
I have pushed a solution and added comments to make it clearer rather than creating another property for the Result class.